File: Bad apple.cpp
#include <iostream>
#include<cmath>
#include <cstdlib>
#include<ctime>
using namespace std;

const int N = 10;
const char marks[2] = {'X', 'O'};
char grid[N][N];
int counter=0;

//This function prints the grid of Tic-Tac-Toe Game as the game progresses
void print_grid() {

    cout << " __";
    for (int i = 0; i < N; cout << "___", i++);
    cout << "   \n";
    cout << "   \n";

    for (int i = 0; i < N; i++) {
        cout << "|  ";
        for (int j = 0; j < N; j++)
            cout << grid[i][j] << "  ";
        cout << "|";
        cout << "  ";
        for (int i = 0; i < N; cout << "  ", i++);
        cout << "  \n";}
     cout << " __";
     for (int i = 0; i < N; cout << "___", i++);
     cout << "   ";

}
//This function checks if the game has a win state or not
bool check_lose(int i , int j ,int a ) {
    if(a==0 || a==1 || a==2){ grid[i][j]='X'; return true;}
    else{ return false;}

}

bool check_win() {
    int c=0;
     for (int i =0 ; i<N; i++){
            for (int j =0 ; j<N; j++){
                if (grid[i][j]=='*'){
                        c+=1;

                }
            }
        }
    if(c==1){return true;}
    else {return false;}

}
bool check_valid_index(int i, int j){
    if(grid[i][j]=='*'){return true;}
    else{return false;}

}


//This function generates pac man
int generate_apples() {
    srand( (unsigned)time( NULL ) );
    int f=pow(N,2);
    int a = rand() % f ;
    return a;
}


//This function checks if given position is valid or not
bool check_valid_position(int i, int j) {
    bool valid= 0<=i && i<N && 0<=j && j<N;
     return valid;

}
//This function sets the given mark to the given cell
void set_cell(int i, int j, int a) {
    if (a%2){grid[i][j]='1'; counter+=1; }
    else if(a%2==0){grid[i][j]='0';}

}
//This function clears the game structures
void grid_clear() {

        for (int i =0 ; i<N; i++){
            for (int j =0 ; j<N; j++){
                grid[i][j]='*';
            }
        }
        counter =0;

}
//This function reads a valid position input
void read_input(int &i, int &j) {
	cout << "Enter the row index and column index: ";
	cin >> i >> j;
    while (!check_valid_position(i, j) || !check_valid_index(i,j)) {
		cout << "Enter a valid row index and a valid column index: ";
		cin >> i >> j;
	}
}
//MAIN FUNCTION
void play_game() {
    cout << " Bad apple Game!\n";
    cout << "Welcome...\n";
    cout << "============================\n";

    while (true) {
        //Prints the grid
        cout<<"Your score is : "<< counter<<endl;
        print_grid();

        //Read an input position from the player
        int i, j;
		read_input(i, j);
		int a = generate_apples();
		if (check_lose(i,j,a)){
            print_grid();
			//Announcement of the final statement
            cout << "Game over !\n";
            break;
		}
		else{
        //Set the player mark in the input position
        set_cell(i, j,a);

        //Check if the grid has a win state
        if (check_win()) {
            //Prints the grid
            print_grid();
			//Announcement of the final statement
            cout << "Congrats, YOU WON";
            break;
        }}

    }
}
int main() {
    while (true) {
    	grid_clear();
    	play_game();
    	char c;
    	cout << "Play Again [Y/N] ";
    	cin >> c;
    	if (c != 'y' && c != 'Y')
    		break;
    }
}

File: Bad apple.py
import random

N= 3
counter = 0
a, b = -1, -1
grid = [['+' for x in range(N)] for y in range(N)]


# This function prints the grid of Gomoku as the game progresses
def print_grid():
    print('--' + '---' * N + '--')
    for i in range(N):
        print(end='|  ')
        for j in range(N):
            print(grid[i][j], end='  ')
        print(end='|')
        print()
    print('--' + '---' * N + '--')


# This function checks if the game has a win state or not
def check_win():
    c = 0
    for i in range(N):
        for j in range(N):
            if grid[i][j] == '*':
                c+=1

    if c == 1:
        return True
    else:
        return False

# This function checks if the game has a tie state or not for the given mark
def check_lose(i,j,a):
    if a == 0 or a == 1 or a == 2:
        grid[i][j]='X'
        return True
    else :
       return False



# This function generates pac man
def generate_apple():
    global a
    a = random.randint(0, N**2)
    return a


def check_valid_index(i,j):
    return  grid[i][j] == '*'

# This function checks if given position is valid or not
def check_valid_position(i,j):

    valid = 0 <=i<N  and   0 <= j <N
    return valid

def set_cell( i,  j,  a):
    global counter
    if a%2:
        grid[i][j]='1'
        counter+=1
    elif a%2==0 :
        grid[i][j]='0'

# This function clears the game structures
def grid_clear():
    global grid
    grid = [['*' for x in range(N)] for y in range(N)]
    global counter
    counter = 0


# This function reads a valid position input
def read_input():
    i ,j =  map(int, input('Enter the row index and column index: ').split())
    while not check_valid_position(i, j) or not check_valid_index(i,j):
        i, j = map(int, input('Enter a valid row index and a valid column index: ').split())
    return i,j


# MAIN FUNCTION
def play_game():
    print("Bad apples Game!")
    print("Welcome...")
    print("============================")
    while True:
        # Prints the grid
        print_grid()
        print("Your score is : ",counter)
        i,j = read_input()
        a = generate_apple()
        # Check if the grid has a tie state
        if check_lose(i,j,a):
            # Prints the grid
            print_grid()
            # Announcement of the final statement
            print("GAME OVER!")
            break
        else:
            set_cell(i,j,a)

        # Check if the state of the grid has a win state
        if check_win():
            # Prints the grid
            print_grid()
            # Announcement of the final statement
            print("Congrats, you won!\n")
            break


while True:
    grid_clear()
    play_game()
    c = input('Play Again [Y/N] ')
    if c not in 'yY':
        break
